tests/Functional/TestCase.php

```diff
@@ -82,16 +86,114 @@ protected static function runFrontScript(array $args, string $cwd = __DIR__): Pr
         return $process;
     }
 
-    protected static function createFiles(array $filenames): void
+    protected static function createFiles(string $baseDir, array $filenames): void
     {
         foreach ($filenames as $filename) {
-            $dirname = dirname($filename);
-            if (!file_exists($dirname)) {
-                self::assertTrue(mkdir($dirname, 0777, true), 'Created directory.');
+            self::createFile($baseDir, $filename);
+        }
+    }
+
+    protected static function createFile(string $baseDir, string $filename): void
+    {
+        $filename = "{$baseDir}/{$filename}";
+        $dirname = dirname($filename);
+        if (!file_exists($dirname)) {
+            self::assertTrue(mkdir($dirname, 0777, true), 'Created directory.');
+        }
+        self::assertTrue(touch($filename), 'Created file.');
+        self::assertNotFalse(realpath($filename), 'Got absolute path.');
+    }
+
+    protected static function changeFile($dir, $filename): void
+    {
+        file_put_contents(
+            DirectoryUtil::ensureTrailingSlash($dir) . $filename,
+            self::CHANGED_CONTENT
+        );
+    }
+
+    protected static function deleteFile($dir, $filename): void
+    {
+        unlink(DirectoryUtil::ensureTrailingSlash($dir) . $filename);
```

Comment:
Any reason not to wrap this in `self::assertTrue()`?

Reply:
Well, I'm wondering how all these assertions in helper code will affect our coverage metrics... but we're already doing it some of the time. May as well be consistent until-and-unless we can confirm that it skews important metrics. Done.